### DISCUSSION TITLE
Добро пожаловать в fct-search обсуждение!

### DISCUSSION THREAD

**Discussion by audetv** [terratensor/svodd#71] [Объявления]
## 👋 Добро пожаловать!
   Обсуждаем идеи и планы по сайту-поиску:
  * Задавайте вопросы.
  * Делитесь идеями.
  * Общайтесь. 

### Новичкам — если не знаете с чего начать, пишите прямо в эту тему!

Краткое руководство по GitHub Discussions. https://docs.github.com/ru/discussions/quickstart

Если у вас есть идея или пожелание, или вы заметили какой-то баг - ошибку или неправильное поведение страниц, сервисов, то можно открыть новую задачу issues и написать об этом в свободной форме тут: https://github.com/audetv/fct-search/issues

По этой ссылке можно посмотреть список закрытых задач и обсуждение по ним:
https://github.com/audetv/fct-search/issues?q=is%3Aissue+is%3Aclosed
Это закрытые [пул запросы](https://docs.github.com/ru/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests):
https://github.com/audetv/fct-search/pulls?q=is%3Apr+is%3Aclosed

Тут можно посмотреть список изменений, которые сделаны на сайте:
https://github.com/audetv/fct-search/commits/main

**Comment by iprst** [comment]
Присоединяюсь! 

Запрос на вытягивание (pull request как его переводит гитхаб) — это допустим, если есть идея что-то сделать, и я могу это сделать, то я «вытягиваю», а потом предлагаю поправленное в основной продукт, верно?

**Reply by audetv** [reply]
Да. так. Разработчик делает копию репозитория `git clone https://github.com/audetv/fct-search.git` у себя на машине, или если это работа с текстами, или какой-то простой  код, то можно прямо в редакторе github'а сделать.
Делаем новую ветку от главной main и в ней работаем. 
Потом делаем коммит - фиксацию изменения и выгружаем git pull. А после уже с помощью интерфейса гитхаба делаем пул запрос на изменение.
 Любой с правом админ или с правом принимать пул запросы в репозиторий может его одобрить и влить в общий код. Перед этим надо проверить, что там сделали в запросе, вытянуть себе эту ветку, переключится на нее, потестировать, «погонять», посмотреть и при необходимости одобрить изменения или написать об изменениях, вплоть до комментирования строк кода, на что обратить внимание.

**Reply by audetv** [reply]
В проекте я настроил actions, автоматический deploy, это значит после вливания кода в main ветку, запускается виртуальная машина на гитхабе, которая собирает сборку, делает новые образы сервисов и потом выгружает их на рабочий сервер. Делает обновления. Это тут https://github.com/audetv/fct-search/actions 
Там же должны запускаться необходимые тесты и проверки и если сборка их проходит, то выходит в продакшн, или получает статус фэйл и сообщение на email отправляет. Это называется CI/CD (Continuous Integration/Continuous Delivery)
Предыдущие сборки сохраняются и в случае необходимости можно откатиться назад.
PS. Если кто-то хочет попробовать себя в программирование, добро пожаловать!

**Reply by audetv** [reply]
А задачи любой из участников может выбрать себе по вкусу из общего списка идей багов и открытых issue. Issue очень удобны при разработке, особенно при групповой или удаленной. Это список задач с приоритетами, образованных из идей, которые стали целью  и т. д. по ПФУ.

**Reply by iprst** [reply]
С программированием знаком лишь едва, могу править какие-то параметры в JS или python, поскольку с ними часто сталкиваюсь. Немного знаю css и html, не слежу за новым функционалом, но какие-то вещи приходилось делать.

**Comment by audetv** [comment]
Краткое руководство по GitHub Discussions. https://docs.github.com/ru/discussions/quickstart
